### PR TITLE
Addition of HTTP method

### PR DESCRIPTION
**Pull request by djzort**
Here is a http plugin function. The user/admin is able to send the log message via HTTP (which could also be called RESTful). Optionally using get, post or push method. Ive used LWP, as HTTP::Tiny doesnt (seem) to have PUT and PUSH functions.

To be really useful, the log program would also be able to read back the logs via HTTP. Also a config file so that logging is quicker, otherwise the command line options become a big burden on making a quick log message!

### CODE REVIEW

**Comment by doherty** [comment]
Thanks for contributing! I also have some comments that I'll put inline on the diffs.

`HTTP::Tiny` can do PUT - and I have never heard of an HTTP verb "PUSH". Would you please look into that? I would prefer this to use `HTTP::Tiny` if possible.

I suppose to read back the log entries, you would need to know something about how they're stored on the other end, and how to retrieve them - so unless this dist also provides the other side of the HTTP connection as well, I'm not sure if that makes sense.

I'd definitely like to have a config file, but I've just never gotten around to it.

File: bin/log [hunk 1]
```diff
@@ -100,6 +111,26 @@ Sets the log directory to use - defaults to F</var/log/sysadmin>.
...
+
+=item * uri - the uri to send to - default is http://localhost
+
+=item * method - the method to send using - default is POST
+
+=back
+
+Use it multiple times to set multiple key-value pairs:
+
+    log --udp-data host=localhost --udp-data port=9002
```
Comment (doherty) [inline bin/log:132]: Should this still say `udp-data`?

File: lib/App/Sysadmin/Log/Simple/Http.pm [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package App::Sysadmin::Log::Simple::Http;
```
Comment (doherty) [inline lib/App/Sysadmin/Log/Simple/Http.pm:1]: Please use caps in the package name - HTTP is an abbreviation.

File: lib/App/Sysadmin/Log/Simple/Http.pm [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    my $ua = LWP::UserAgent->new;
+    $ua->timeout($HTTP_TIMEOUT);
+    my $res;
+
+    if ( lc $self->{http}->{method} eq 'get' ) {
+
+        my $uri = $self->{http}->{uri};
+        $uri .= ($uri =~ m/\?/) ? '&' : '?';
+        $uri .= sprintf('user=%s&log=%s',$self->{user},$logentry); #FIXME these need to be escaped
```
Comment (doherty) [inline lib/App/Sysadmin/Log/Simple/Http.pm:79]: Yes, these need to be escaped `:)`

File: lib/App/Sysadmin/Log/Simple/Http.pm [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+        $res = $ua->get($uri);
+
+    } elsif ( lc $self->{http}->{method} eq 'post' ) {
```
Comment (doherty) [inline lib/App/Sysadmin/Log/Simple/Http.pm:83]: Could you please un-cuddle these to match the code style in the rest of the dist?

Comment (djzort) [inline lib/App/Sysadmin/Log/Simple/Http.pm:83]: will do

File: lib/App/Sysadmin/Log/Simple/Http.pm (hunk 2 above)
Comment (djzort) [inline lib/App/Sysadmin/Log/Simple/Http.pm:1]: will do

File: bin/log (hunk 1 above)
Comment (djzort) [inline bin/log:132]: yup, copy and paste boo boo

**Comment by djzort** [comment]
yeah sorry, push was a typo. looking in the perldoc of HTTP::Tiny i dont see anything except for 'get' ? anyway i will close this request and send another one with the changes as requested.

**Pull request closed without merging**